### PR TITLE
upgrade ember-cli to 2.12.3

### PR DESCRIPTION
**Pull request by helloWorldCindy**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

Upgrade ember-cli to 2.12.3

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -8,13 +9,12 @@
...
 .ember-cli
 .eslintignore
-.eslintrc
 .github
 .gitignore
-.jshintrc
 .pullapprove.yml
 .travis
 .watchmanconfig
+.eslintrc.js
```
Comment (notmessenger) [inline .npmignore]: Locate in file alphabetically

File: .travis.yml [hunk 2]
```diff
@@ -48,6 +33,7 @@ matrix:
   allow_failures:
   - env: EMBER_TRY_SCENARIO=ember-beta
   - env: EMBER_TRY_SCENARIO=ember-canary
+  - node_js: 'stable'
```
Comment (notmessenger) [inline .travis.yml]: Same question as other PRs @juwara0 about whether needed and that I haven't bee doing this and whether should be.

File: package.json [hunk 3]
```diff
@@ -23,10 +23,9 @@
   "author": "Matthew Dahl (https://github.com/sandersky)",
   "license": "MIT",
   "devDependencies": {
```
Comment (notmessenger) [inline package.json:25]: `ember-cli-test-loader` should be removed

File: package.json [hunk 4]
```diff
@@ -39,12 +38,12 @@
     "ember-cli-uglify": "^1.2.0",
     "ember-disable-prototype-extensions": "^1.1.0",
     "ember-export-application-global": "^1.1.1",
-    "ember-load-initializers": "^0.6.3",
+    "ember-load-initializers": "^0.6.0",
     "ember-lodash-shim": "^2.0.0",
-    "ember-resolver": "^2.1.1",
+    "ember-resolver": "^2.0.3",
     "ember-source": "~2.11.2",
```
Comment (notmessenger) [inline package.json]: Should be `~2.12.0`

File: tests/.eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  env: {
+    embertest: true
+  }
+};
```
Comment (notmessenger) [inline tests/.eslintrc.js]: Remove `;`

File: tests/dummy/config/environment.js [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/* eslint-env node */
 module.exports = function (environment) {
```
Comment (notmessenger) [inline tests/dummy/config/environment.js:2]: `ENV.rootURL = '/'` in test configuration should be removed as well

Comment (notmessenger) [reply tests/dummy/config/environment.js:2]: Change `locationType` to `hash` for consistency

**Comment by notmessenger** [comment]
Change the branch names in:

* https://github.com/ciena-blueplanet/ember-graphlib/blob/master/.travis/maybe-bump-version.sh#L9
* https://github.com/ciena-blueplanet/ember-graphlib/blob/master/.travis/maybe-publish-coverage.sh#L17

**Review by notmessenger — CHANGES_REQUESTED** [review]

File: .npmignore (hunk 1 above)
Comment (helloWorldCindy) [reply .npmignore]: Fixed.

File: package.json (hunk 3 above)
Comment (helloWorldCindy) [reply package.json:25]: Fixed.

File: package.json (hunk 4 above)
Comment (helloWorldCindy) [reply package.json]: Fixed.

File: tests/.eslintrc.js (hunk 5 above)
Comment (helloWorldCindy) [reply tests/.eslintrc.js]: Fixed.

File: tests/dummy/config/environment.js (hunk 6 above)
Comment (helloWorldCindy) [reply tests/dummy/config/environment.js:2]: Fixed.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -1,9 +1,9 @@
-**This project uses [semver](http://semver.org), please check the scope of this pr:**
+### This project uses [semver](semver.org), please check the scope of this pr:
```
Comment (dafortin) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: @juwara0 @notmessenger we should agree on ** or ###. `ember-frost-core` is using ** and `ember-frost-list` is using `###`.

File: .travis.yml [hunk 8]
```diff
@@ -14,28 +14,16 @@ addons:
...
-  - EMBER_TRY_SCENARIO=ember-2-6
-  - EMBER_TRY_SCENARIO=ember-2-7
-  - EMBER_TRY_SCENARIO=ember-2-8
-  - EMBER_TRY_SCENARIO=ember-2-9
-  - EMBER_TRY_SCENARIO=ember-2-10
-  - EMBER_TRY_SCENARIO=default
+  - EMBER_TRY_SCENARIO=ember-lts-2.4
+  - EMBER_TRY_SCENARIO=ember-lts-2.8
+  - EMBER_TRY_SCENARIO=ember-default
   - EMBER_TRY_SCENARIO=ember-release
```
Comment (dafortin) [inline .travis.yml:27]: We decided to keep `release` `beta` and `canary` for this repo?

Comment (juwara0) [reply .travis.yml:27]: Yes, for the shims I have expanded the build matrix.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (notmessenger) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Agreed though I'm not concerned with making that determination for the sake of this PR.  Consistency between the repos will be addressed in a separate effort.

(if there was an obvious pattern and this deviated, yes - it should be fixed now.  but to use sidetrack this PR to determine a desired pattern is not anything we need to do right now)

Comment (juwara0) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Just a quick look through the repos and I am see more `**`. We can address a consistent approach across the repos in a separate effort.

File: .npmignore [hunk 9]
```diff
@@ -1,17 +1,17 @@
...
 /config
+/config/ember-try.js
 /dist
 /tests
 /tmp
 **/.gitkeep
 .bowerrc
 .editorconfig
 .ember-cli
+.eslintrc.js
```
Comment (notmessenger) [inline .npmignore]: Still isn't alphabetical :-)

Comment (juwara0) [reply .npmignore]: Fixed

**Review by notmessenger — APPROVED** [review]

**Comment by notmessenger** [comment]
👍

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (dafortin) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: I'm good with this. As long as we are aware that we need to fix that later. (it's a very minor thing)

**Pull request merged**